app/operations/fdsh/ridp/rj139/handle_primary_determination_request.rb

```diff
@@ -0,0 +1,196 @@
+
+          if result.success?
+            @response_transmission = result.value!
+            Success(@response_transmission)
+          else
+            add_errors({ transaction: @request_transaction, transmission: @request_transmission, job: @job },
+                       "Failed to create response transmission due to #{result.failure}",
+                       :create_response_transmission)
+            status_result = update_status({ job: @job }, :failed, "Failed to create response transmission")
+            return status_result if status_result.failure?
+            result
+          end
+        end
+
+        def create_response_transaction(values, ridp_response)
+          subject = values[:transaction].transactable
+          result = Fdsh::Jobs::CreateTransaction.new.call(values.merge({ key: :ridp_primary_verification_response,
+                                                                         started_at: DateTime.now,
+                                                                         transmission: @response_transmission,
+                                                                         title: 'RIDP Primary Verification Response',
+                                                                         description: 'Response for primary RIDP verification from CMS',
+                                                                         subject: subject,
+                                                                         event: 'received',
+                                                                         state_key: :received }))
+
+          if result.success?
+            @response_transaction = result.value!
+            update_response_transmission(ridp_response)
+            Success(@response_transaction)
+          else
+            add_errors({ transaction: @request_transaction, transmission: @request_transmission, job: @job },
+                       "Failed to create response transaction due to #{result.failure}",
+                       :create_response_transaction)
+            status_result = update_status({ transmission: @response_transmission, job: @job }, :failed, "Failed to create response transaction")
+            return status_result if status_result.failure?
+            result
+          end
+        end
+
+        def update_response_transmission(ridp_response)
+          parsed_payload = JSON.parse(ridp_response.env.response_body)
+          @response_transaction.json_payload = parsed_payload
+          @final_decision = parsed_payload.dig("ridpResponse", "finalDecisionCode")
+          unless @final_decision
+            @response_transaction.metadata = { session_id: parsed_payload['ridpResponse']['sessionIdentification'] }
+            @job.title = "RIDP Primary Request for #{parsed_payload['ridpResponse']['sessionIdentification']}"
+          end
+          @response_transaction.save
```

Comment:
Pushed change to add a check for if it is not json, in which case it will return a failure message and save the error message to the transmittable objects